webrecorder/browsertrix
# Paginate Workflows and refactor to use server-side queries

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/859



### Changes

- Paginates Crawl Workflows when there are more than 10 workflows
- Refactors workflow search and crawl search to use the same component
- Adds sort by first seed, workflow creation date, and workflow modified date
- Separates "last run" date from "modified" date

### Manual testing

1. Log in as a superadmin user and click an org with >10 workflows
2. Click pagination controls. Verify list paginates as expected
3. Enter text into search field. Verify search functions the same as searching for crawls in "Archived Items"
4. Sort by options and reverse sort. Verify list is ordered as expected


### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Crawling | **Pagination:**<br/><img width="304" alt="Screenshot 2023-08-21 at 4 09 35 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/2b0bb836-09d5-4b3b-aa07-bad90c972cbc"><br/><br/>**Search values:**<br/><img width="726" alt="Screenshot 2023-08-21 at 4 15 47 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/f7966bca-5036-47d0-932d-ca40cd930194"><br/><br/>**Headers & Row**:<br /><img width="1109" alt="Screenshot 2023-08-22 at 11 02 12 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/a30832f7-ccac-43ab-a0b5-277fdecbe929"> |
